lib/src/boilerplate_suggestors/props_mixins_migrator.dart

```diff
@@ -16,12 +16,78 @@ import 'package:analyzer/dart/ast/ast.dart';
 import 'package:analyzer/dart/ast/visitor.dart';
 import 'package:codemod/codemod.dart';
 
+import 'boilerplate_utilities.dart';
+
 /// Suggestor that updates stand alone props mixins to be actual mixins.
+///
+/// > Related: [PropsMetaMigrator]
 class PropsMixinMigrator extends GeneralizingAstVisitor
     with AstVisitingSuggestorMixin
     implements Suggestor {
   @override
   visitClassDeclaration(ClassDeclaration node) {
     super.visitClassDeclaration(node);
+
+    if (!shouldMigratePropsAndStateMixin(node)) return;
+
+    migrateClassToMixin(node, yieldPatch);
+    _removePropsOrStateGetter(node);
+    _removePropsOrStateMixinAnnotation(node);
+    _migrateMixinMetaField(node);
+  }
+
+  void _removePropsOrStateGetter(ClassDeclaration node) {
+    final classGetters = node.members
+        .whereType<MethodDeclaration>()
+        .where((method) => method.isGetter);
+    final propsOrStateGetter = classGetters.singleWhere(
+        (getter) => getter.name.name == 'props' || getter.name.name == 'state',
+        orElse: () => null);
+    if (propsOrStateGetter != null) {
+      yieldPatch(propsOrStateGetter.offset, propsOrStateGetter.end + 1, '');
```

Comment:
Magic number: what is the `1` for? If it's for the newline, that isn't necessary since we'll be running dartfmt afterwards (and our test cases are format-agnostic as a result of the inputs and outputs being formatted before comparison).